### ISSUE TITLE
Customer subscribed online does not receive messages only when reconnecting

### ISSUE THREAD

**Issue by keniodev** [vernemq/vernemq#2365]
### Describe your issue or feature request

I'm running a verne cluster with 2 nodes, they connect and send messages between them, but when I send, the subscriber I don't receive it immediately only when I reconnect with it.

this is my configuration file:

allow_anonymous = off
allow_register_during_netsplit = off
allow_publish_during_netsplit = off
allow_subscribe_during_netsplit = off
allow_unsubscribe_during_netsplit = off
allow_multiple_sessions = off
coordinate_registrations = on
max_inflight_messages = 20
max_online_messages = 1000
max_offline_messages = 1000
override_max_online_messages = off
max_message_size = 0
upgrade_outgoing_qos = off
listener.max_connection_lifetime = 0
listener.max_connections = 10000
max_ws_frame_size = 268435456
listener.nr_of_acceptors = 10
listener.tcp.default = 0.0.0.0:1884
listener.tcp.name.allow_anonymous_override = off
listener.ssl.name.allow_anonymous_override = off
listener.vmq.clustering = 0.0.0.0:44054
listener.http.default = 127.0.0.1:8888
listener.ssl.pskfile = /etc/vernemq/vmq.psk
systree_enabled = on
systree_interval = 20000
graphite_enabled = off
graphite_host = localhost
graphite_port = 2003
graphite_interval = 20000
shared_subscription_policy = prefer_local
plugins.vmq_passwd = on
plugins.vmq_acl = on
plugins.vmq_diversity = off
plugins.vmq_webhooks = off
plugins.vmq_bridge = off
topic_max_depth = 10
metadata_plugin = vmq_swc
vmq_acl.acl_file = /etc/vernemq/vmq.acl
vmq_acl.acl_reload_interval = 10
vmq_passwd.password_file = /etc/vernemq/vmq.passwd
vmq_passwd.password_reload_interval = 10
vmq_diversity.script_dir = /usr/share/vernemq/lua
vmq_diversity.auth_postgres.enabled = off
vmq_diversity.postgres.ssl = off
vmq_diversity.postgres.password_hash_method = crypt
vmq_diversity.auth_cockroachdb.enabled = off
vmq_diversity.cockroachdb.ssl = on
vmq_diversity.cockroachdb.password_hash_method = bcrypt
vmq_diversity.auth_mysql.enabled = off
vmq_diversity.mysql.password_hash_method = password
vmq_diversity.auth_mongodb.enabled = off
vmq_diversity.mongodb.ssl = off
vmq_diversity.auth_redis.enabled = off
vmq_bcrypt.pool_size = 1
vmq_bcrypt.nif_pool_size = 4
vmq_bcrypt.nif_pool_max_overflow = 10
vmq_bcrypt.default_log_rounds = 12
vmq_bcrypt.mechanism = port
log.console = file
log.console.level = info
log.console.file = /var/log/vernemq/console.log
log.error.file = /var/log/vernemq/error.log
log.syslog = off
log.crash = on
log.crash.file = /var/log/vernemq/crash.log
log.crash.maximum_message_size = 64KB
log.crash.size = 10MB
log.crash.rotation = $D0
log.crash.rotation.keep = 5
nodename = vernemq2@111.111.111.111
distributed_cookie = vmq
erlang.async_threads = 64
erlang.max_ports = 262144
leveldb.maximum_memory.percent = 70
leveldb.compression = on
leveldb.compression.algorithm = lz4
include conf.d/*.conf

All ports are open and I've already done a ping test

### Describe the solution you'd like

Orientação sobre alguma configuração que possivelmente não encontrei na documentação ou não me atentei

### Describe alternatives you've considered

Guidance on some configuration that I possibly didn't find in the documentation or didn't pay attention to

### Environment

- VerneMQ Version: 1.13.0
- OS: Ubuntu 22.04
- Erlang/OTP version (if building from source): 12.2.1
- Cluster size/standalone: 1 nó

### Add any other context, code samples or screenshots about the issue report here.

```markdown
sudo tail -f /var/log/vernemq/console.log
2024-10-29 20:53:36.817 [warning] <0.619.0>@vmq_cluster_node:handle_message:{255,5} connection to node 'vernemq2@111.111.111.111' has been closed, reconnect in 1000ms
2024-10-29 20:53:37.819 [info] <0.619.0>@vmq_cluster_node:handle_message:{218,13} successfully connected to cluster node 'vernemq2@111.111.111.111'
2024-10-29 20:53:37.819 [warning] <0.14393.0>@vmq_cluster_com:loop:{101,14} terminate due to remote_node_not_available
```

### Code of Conduct

- [X] I agree to follow the VerneMQ's Code of Conduct

**Comment by ioolkos** [comment]
@keniodev your log lines show a reconnect of the message-transporting connection between the nodes which can happen sometimes. Check whether this is a one-time instance or happening constantly.
I don't think this is necessarily related to the consumer only getting a message after re-connect. Maybe you can experiment a bit more and generate better insights.

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by keniodev** [comment]
@ioolkos This happens constantly, the log shown is only a part but it keeps repeating itself in a loop.

All the necessary doors are open, but I still don't understand what is happening

**Comment by ioolkos** [comment]
@keniodev Can you set `outgoing_clustering_buffer_size = 1000000` (1MB) in the `vernemq.conf` file? if this doesn't help, we can also increase the TCP buffers of those connections. 

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by keniodev** [comment]
@ioolkos  I discovered the error I was making, because I was following the installation recommendations from a website and had the vmq.args file. Well, I reset the VPS and started the installation process again without this file, and in `tcp.name = <IP_LOCAL>:<PORT>` and `vmq.clustering = <IP_LOCAL>:<PORT>`. After performing the clustering, the messages are now updated in real time.

I'm going to close, thanks for your attention!
